Building Qiskit 2.5.0 C tests fails when using extrict build flags

### Environment

- **Qiskit version**:  2.5.0
- **Python version**: 3.14.6
- **Operating system**:  Arch Linux


### What is happening?

When running ~`make c`~ `make ctest` in the new release 2.5.0, the build fails with the following errors:

```console
make[1]: Entering directory '/build/python-qiskit/src/qiskit/test/c/build'
[  2%] Building C object CMakeFiles/common.dir/common.c.o
[  5%] Linking C static library libcommon.a
[  5%] Built target common
[  8%] Building C object CMakeFiles/test_driver.dir/test_driver.c.o
[ 11%] Building C object CMakeFiles/test_driver.dir/test_classical_expr.c.o
[ 14%] Building C object CMakeFiles/test_driver.dir/test_control_flow.c.o
[ 17%] Building C object CMakeFiles/test_driver.dir/test_elide_permutations.c.o
[ 20%] Building C object CMakeFiles/test_driver.dir/test_consolidate_blocks.c.o
[ 23%] Building C object CMakeFiles/test_driver.dir/test_inverse_cancellation.c.o
[ 26%] Building C object CMakeFiles/test_driver.dir/test_basis_translator.c.o
[ 32%] Building C object CMakeFiles/test_driver.dir/test_commutative_cancellation.c.o
[ 32%] Building C object CMakeFiles/test_driver.dir/test_optimize_1q_sequences.c.o
[ 35%] Building C object CMakeFiles/test_driver.dir/test_gate_direction.c.o
[ 38%] Building C object CMakeFiles/test_driver.dir/test_dag.c.o
[ 41%] Building C object CMakeFiles/test_driver.dir/test_neighbors.c.o
[ 44%] Building C object CMakeFiles/test_driver.dir/test_iqp.c.o
[ 47%] Building C object CMakeFiles/test_driver.dir/test_param.c.o
[ 50%] Building C object CMakeFiles/test_driver.dir/test_circuit.c.o
[ 52%] Building C object CMakeFiles/test_driver.dir/test_remove_diagonal_gates_before_measure.c.o
[ 55%] Building C object CMakeFiles/test_driver.dir/test_pbc.c.o
[ 61%] Building C object CMakeFiles/test_driver.dir/test_remove_identity_equiv.c.o
[ 61%] Building C object CMakeFiles/test_driver.dir/test_quantum_volume.c.o
[ 64%] Building C object CMakeFiles/test_driver.dir/test_complex.c.o
[ 67%] Building C object CMakeFiles/test_driver.dir/test_sabre_layout.c.o
[ 70%] Building C object CMakeFiles/test_driver.dir/test_sparse_observable.c.o
[ 73%] Building C object CMakeFiles/test_driver.dir/test_split_2q_unitaries.c.o
[ 76%] Building C object CMakeFiles/test_driver.dir/test_suzuki_trotter.c.o
/build/python-qiskit/src/qiskit/test/c/test_classical_expr.c: In function ‘test_op_types_roundtrip’:
/build/python-qiskit/src/qiskit/test/c/test_classical_expr.c:166:43: error: implicit conversion from ‘QkUnaryOpType’ to ‘QkBinaryOpType’ [-Werror=enum-conversion]
  166 |         expr = inner_test_binary_expr_ops(expected_op);
      |                                           ^~~~~~~~~~~
/build/python-qiskit/src/qiskit/test/c/test_classical_expr.c:176:28: error: comparison between ‘QkBinaryOpType’ and ‘QkUnaryOpType’ [-Werror=enum-compare]
  176 |         if (binary_info.op != expected_op) {
      |                            ^~
[ 79%] Building C object CMakeFiles/test_driver.dir/test_target.c.o
/build/python-qiskit/src/qiskit/test/c/test_gate_direction.c: In function ‘create_target’:
/build/python-qiskit/src/qiskit/test/c/test_gate_direction.c:26:68: error: comparison between ‘QkExitCode’ and ‘enum TestResult’ [-Werror=enum-compare]
   26 |     if (qk_target_entry_add_property(cx_entry, qargs, 2, 0.0, 0.0) != Ok ||
      |                                                                    ^~
/build/python-qiskit/src/qiskit/test/c/test_gate_direction.c:27:72: error: comparison between ‘QkExitCode’ and ‘enum TestResult’ [-Werror=enum-compare]
   27 |         qk_target_entry_add_property(cx_entry, &qargs[1], 2, 0.0, 0.0) != Ok ||
      |                                                                        ^~
/build/python-qiskit/src/qiskit/test/c/test_gate_direction.c:28:53: error: comparison between ‘QkExitCode’ and ‘enum TestResult’ [-Werror=enum-compare]
   28 |         qk_target_add_instruction(target, cx_entry) != Ok ||
      |                                                     ^~
/build/python-qiskit/src/qiskit/test/c/test_gate_direction.c:29:69: error: comparison between ‘QkExitCode’ and ‘enum TestResult’ [-Werror=enum-compare]
   29 |         qk_target_entry_add_property(rzx_entry, qargs, 2, 0.0, 0.0) != Ok ||
      |                                                                     ^~
/build/python-qiskit/src/qiskit/test/c/test_gate_direction.c:30:54: error: comparison between ‘QkExitCode’ and ‘enum TestResult’ [-Werror=enum-compare]
   30 |         qk_target_add_instruction(target, rzx_entry) != Ok) {
      |                                                      ^~
/build/python-qiskit/src/qiskit/test/c/test_gate_direction.c: In function ‘test_standalone_check_gate_direction’:
/build/python-qiskit/src/qiskit/test/c/test_gate_direction.c:51:17: error: implicit conversion from ‘QkExitCode’ to ‘enum TestResult’ [-Werror=enum-conversion]
   51 |     if ((result = qk_circuit_gate(circuit, QkGate_CX, qargs, NULL)) != Ok ||
      |                 ^
/build/python-qiskit/src/qiskit/test/c/test_gate_direction.c:52:17: error: implicit conversion from ‘QkExitCode’ to ‘enum TestResult’ [-Werror=enum-conversion]
   52 |         (result = qk_circuit_gate(circuit, QkGate_CX, &qargs[1], NULL)) != Ok) {
      |                 ^
/build/python-qiskit/src/qiskit/test/c/test_gate_direction.c:61:21: error: implicit conversion from ‘QkExitCode’ to ‘enum TestResult’ [-Werror=enum-conversion]
   61 |         if ((result = qk_circuit_gate(circuit, QkGate_CX, &qargs[2], NULL)) != Ok) {
      |                     ^
/build/python-qiskit/src/qiskit/test/c/test_gate_direction.c: In function ‘test_standalone_gate_direction_simple’:
/build/python-qiskit/src/qiskit/test/c/test_gate_direction.c:91:17: error: implicit conversion from ‘QkExitCode’ to ‘enum TestResult’ [-Werror=enum-conversion]
   91 |     if ((result = qk_circuit_gate(circuit, QkGate_CX, qargs, NULL)) != Ok ||     // stays as is
      |                 ^
/build/python-qiskit/src/qiskit/test/c/test_gate_direction.c:92:17: error: implicit conversion from ‘QkExitCode’ to ‘enum TestResult’ [-Werror=enum-conversion]
   92 |         (result = qk_circuit_gate(circuit, QkGate_CX, &qargs[1], NULL)) != Ok || // stays as is
      |                 ^
/build/python-qiskit/src/qiskit/test/c/test_gate_direction.c:93:17: error: implicit conversion from ‘QkExitCode’ to ‘enum TestResult’ [-Werror=enum-conversion]
   93 |         (result = qk_circuit_gate(circuit, QkGate_CX, &qargs[2], NULL)) !=
      |                 ^
/build/python-qiskit/src/qiskit/test/c/test_gate_direction.c:95:17: error: implicit conversion from ‘QkExitCode’ to ‘enum TestResult’ [-Werror=enum-conversion]
   95 |         (result = qk_circuit_gate(circuit, QkGate_RZX, &qargs[3], params)) !=
      |                 ^
[ 82%] Building C object CMakeFiles/test_driver.dir/test_transpile_layout.c.o
[ 85%] Building C object CMakeFiles/test_driver.dir/test_unitary_synthesis.c.o
[ 88%] Building C object CMakeFiles/test_driver.dir/test_transpiler.c.o
[ 91%] Building C object CMakeFiles/test_driver.dir/test_version_info.c.o
[ 94%] Building C object CMakeFiles/test_driver.dir/test_two_qubit_peephole.c.o
[ 97%] Building C object CMakeFiles/test_driver.dir/test_vf2_layout.c.o
cc1: all warnings being treated as errors
make[3]: *** [CMakeFiles/test_driver.dir/build.make:219: CMakeFiles/test_driver.dir/test_gate_direction.c.o] Error 1
make[3]: *** Waiting for unfinished jobs....
cc1: all warnings being treated as errors
make[3]: *** [CMakeFiles/test_driver.dir/build.make:121: CMakeFiles/test_driver.dir/test_classical_expr.c.o] Error 1
make[2]: *** [CMakeFiles/Makefile2:122: CMakeFiles/test_driver.dir/all] Error 2
make[1]: *** [Makefile:101: all] Error 2
make[1]: Leaving directory '/build/python-qiskit/src/qiskit/test/c/build'
make: *** [Makefile:165: ctest] Error 2
```

This is due to stricter compile flags set by Arch Linux:

```console
#-- Compiler and Linker Flags
#CPPFLAGS=""
CFLAGS="-march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions \
        -Wp,-D_FORTIFY_SOURCE=3 -Wformat -Werror=format-security \
        -fstack-clash-protection -fcf-protection \
        -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer"
CXXFLAGS="$CFLAGS -Wp,-D_GLIBCXX_ASSERTIONS"
LDFLAGS="-Wl,-O1 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now \
         -Wl,-z,pack-relative-relocs"
LTOFLAGS="-flto=auto"
```

Probably only shown as warnings in the GitHub CI.

### How can we reproduce the issue?

Build with stricter flags as mentioned above.

### What should happen?

Should build as with 2.4.x.

### Any suggestions?

New versions should not introduce additional warnings. Maybe it would be useful to make the CI build pipeline fail instead of letting warnings pass.

--- a/test/c/test_classical_expr.c
+++ b/test/c/test_classical_expr.c
@@ -161,7 +161,7 @@ static int test_op_types_roundtrip(void) {
     }
 
     for (size_t op = QkBinaryOpType_BitAnd; op <= QkBinaryOpType_Div; op++) {
-        QkUnaryOpType expected_op = (QkUnaryOpType)op;
+        QkBinaryOpType expected_op = (QkBinaryOpType)op;
 
         expr = inner_test_binary_expr_ops(expected_op);
 

--- a/test/c/test_gate_direction.c
+++ b/test/c/test_gate_direction.c
@@ -23,11 +23,11 @@ static QkTarget *create_target() {
     double rzx_params[1] = {1.5};
     QkTargetEntry *rzx_entry = qk_target_entry_new_fixed(QkGate_RZX, rzx_params, "rzx");
 
-    if (qk_target_entry_add_property(cx_entry, qargs, 2, 0.0, 0.0) != Ok ||
-        qk_target_entry_add_property(cx_entry, &qargs[1], 2, 0.0, 0.0) != Ok ||
-        qk_target_add_instruction(target, cx_entry) != Ok ||
-        qk_target_entry_add_property(rzx_entry, qargs, 2, 0.0, 0.0) != Ok ||
-        qk_target_add_instruction(target, rzx_entry) != Ok) {
+    if (qk_target_entry_add_property(cx_entry, qargs, 2, 0.0, 0.0) != QkExitCode_Success ||
+        qk_target_entry_add_property(cx_entry, &qargs[1], 2, 0.0, 0.0) != QkExitCode_Success ||
+        qk_target_add_instruction(target, cx_entry) != QkExitCode_Success ||
+        qk_target_entry_add_property(rzx_entry, qargs, 2, 0.0, 0.0) != QkExitCode_Success ||
+        qk_target_add_instruction(target, rzx_entry) != QkExitCode_Success) {
         printf("Unexpected error encountered in create_target.");
         qk_target_free(target);
         return NULL;
@@ -44,12 +44,12 @@ static int test_standalone_check_gate_direction(void) {
     if (!target)
         return RuntimeError;
 
-    enum TestResult result = Ok;
+    int result = Ok;
     QkCircuit *circuit = qk_circuit_new(3, 0);
     uint32_t qargs[4] = {0, 1, 2, 1};
 
-    if ((result = qk_circuit_gate(circuit, QkGate_CX, qargs, NULL)) != Ok ||
-        (result = qk_circuit_gate(circuit, QkGate_CX, &qargs[1], NULL)) != Ok) {
+    if ((result = qk_circuit_gate(circuit, QkGate_CX, qargs, NULL)) != QkExitCode_Success ||
+        (result = qk_circuit_gate(circuit, QkGate_CX, &qargs[1], NULL)) != QkExitCode_Success) {
         printf("Unexpected error encountered while adding CX gates in test_check_gate_direction.");
         goto cleanup;
     }
@@ -58,7 +58,7 @@ static int test_standalone_check_gate_direction(void) {
     if (!check_pass)
         result = EqualityError;
     else {
-        if ((result = qk_circuit_gate(circuit, QkGate_CX, &qargs[2], NULL)) != Ok) {
+        if ((result = qk_circuit_gate(circuit, QkGate_CX, &qargs[2], NULL)) != QkExitCode_Success) {
             printf("Unexpected error encountered while adding a CX gate in "
                    "test_check_gate_direction.");
             goto cleanup;
@@ -82,18 +82,20 @@ static int test_standalone_gate_direction_simple(void) {
     if (!target)
         return RuntimeError;
 
-    enum TestResult result = Ok;
+    int result = Ok;
 
     QkCircuit *circuit = qk_circuit_new(3, 0);
     uint32_t qargs[5] = {0, 1, 2, 1, 0};
     double params[1] = {1.5};
 
-    if ((result = qk_circuit_gate(circuit, QkGate_CX, qargs, NULL)) != Ok ||     // stays as is
-        (result = qk_circuit_gate(circuit, QkGate_CX, &qargs[1], NULL)) != Ok || // stays as is
+    if ((result = qk_circuit_gate(circuit, QkGate_CX, qargs, NULL)) !=
+            QkExitCode_Success || // stays as is
+        (result = qk_circuit_gate(circuit, QkGate_CX, &qargs[1], NULL)) !=
+            QkExitCode_Success || // stays as is
         (result = qk_circuit_gate(circuit, QkGate_CX, &qargs[2], NULL)) !=
-            Ok || // would be replaced by 5 gates
+            QkExitCode_Success || // would be replaced by 5 gates
         (result = qk_circuit_gate(circuit, QkGate_RZX, &qargs[3], params)) !=
-            Ok) { // would be replaced by 5 gates
+            QkExitCode_Success) { // would be replaced by 5 gates
         printf("Unexpected error encountered while adding gates in test_gate_direction.");
         goto cleanup;
     }

